### PR TITLE
Adding pypolymlp as a package

### PR DESCRIPTION
**Pull request by sekocha**
`@conda-forge/help-python-c, ready for review!`

Checklist
- [ x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x ] Source is from official source.
- [x ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x ] If static libraries are linked in, the license of the static library is packaged.
- [x ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x ] Build number is 0.
- [ x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sekocha** [comment]
@conda-forge/help-python-c, ready for review!

**Comment by sekocha** [comment]
@conda-forge/help-python-c, ready for review!
@conda-forge/staged-recipes

I appreciate it if pypolymlp would be reviewed as a package.
Thank you very much.

Checklist

[ x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
[ x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
[x ] Source is from official source.
[x ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
[x ] If static libraries are linked in, the license of the static library is packaged.
[x ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
[x ] Build number is 0.
[ x] A tarball (url) rather than a repo (e.g. git_url) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
[ x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
[ x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

**Comment by sekocha** [comment]
@conda-forge/help-python-c, @conda-forge/help-python-c, ready for review!

File: recipes/pypolymlp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - pypolymlp-structure=pypolymlp.api.run_polymlp_str:run
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - ninja
+    - wheel
```
Comment (xhochy) [inline recipes/pypolymlp/meta.yaml:27]: This should be in host

File: recipes/pypolymlp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - cmake
+    - make
+    - pybind11
+    - pybind11-global
```
Comment (xhochy) [inline recipes/pypolymlp/meta.yaml:31]: Move to host

File: recipes/pypolymlp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - eigen
+    - libgomp  # [linux]
+    - llvm-openmp  # [osx]
```
Comment (xhochy) [inline recipes/pypolymlp/meta.yaml:34]: Please add them also to host, i.e. duplicate them.

File: recipes/pypolymlp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    - python
+    - setuptools >=61.0
+    - scikit-build-core
+    - setuptools_scm
+    - pip
+  run:
+    - python
+    - scipy
+    - pyyaml
+    - {{ pin_compatible('numpy') }}
```
Comment (xhochy) [inline recipes/pypolymlp/meta.yaml:45]: numpy is not in host, thus the pin_compatible function has no effect.

File: recipes/pypolymlp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    - setuptools >=61.0
+
+test:
+  imports:
+    - pypolymlp
+  commands:
```
Comment (xhochy) [inline recipes/pypolymlp/meta.yaml:51]:
```suggestion
test:
  imports:
    - pypolymlp
  requires:
    - pip
  commands:
    - pip check
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by sekocha** [comment]
Thank you for taking the time to review the package. Following the requests, I have revised meta.yaml.

**Review by xhochy — APPROVED** [review]

**Pull request merged**